build-system/pr-check/npm-checks.js

```diff
@@ -65,10 +66,13 @@ function isPackageLockFileInSync() {
 }
 
 /**
- * Makes sure that package-lock.json was properly updated.
+ * Makes sure that package-lock.json was properly updated after running `npm i`,
+ * which will reveal any missing changes. (We do this in a separate step because
+ * the clean-install command `npm ci` doesn't update the lockfile.)
  * @return {boolean}
  */
 function isPackageLockFileProperlyUpdated() {
+  exec('npm install', {'stdio': 'ignore'});
   const filesChanged = gitDiffNameOnly();
   const loggingPrefix = getLoggingPrefix();
```

Comment:
Can we add an explicit check that `JSON.parse(readFile('package-lock.json')).lockfileVersion === 1`?

Reply:
A check for `lockfileVersion == 1` is built in to this change because we explicitly install `node` LTS during setup:

https://github.com/ampproject/amphtml/blob/2f932e6ee6f68c203268f897e53e9dbde08cd175/.circleci/install_dependencies.sh#L31-L32

And since we're running `npm install` before verifying, `lockfileVersion` will be overwritten to `1` even if someone tries to change it, and the presence of diffs will fail CI.

Reply:
Yes, but the messaging we're using isn't clear, and it's because we can't tell between a missing package and an incorrect npm version. Having an explicit check would allow us to output a clear error.

Reply:
Gotcha, I can add a check. We'll have to verify `lockfileVersion` first, then run `npm install`, and then check for other unexpected diffs. Coming right up.